components/com_content/src/Model/CategoryModel.php

```diff
@@ -335,7 +335,7 @@ public function getPagination()
     public function getCategory()
     {
         if (!\is_object($this->_item)) {
-            if (!empty($this->state->get('params'))) {
+            if (isset($this->state) && !empty($this->state->get('params'))) {
```

Comment:
It works but maybe we can shorter code using Null-safe operator
```suggestion
            if (!empty($this->state?->get('params'))) {
```

Reply:
Shorter and modern.

Reply:
At the end it's a matter of taste. I'm ok with both.

Reply:
@heelc29 So I leave it on you to decide if you want to follow @joomdonation ' suggestion or not. If yes, just apply it and I will restore @chmst ' test result, and it no, then just resolve this conversation.

To me it's a matter of taste, both has advantages and disadvantages, and our code is not consistently using the one or the other way. It could also be done with a future PR to change all the statements of this kind to use null-safe `? ->`.

Reply:
@richard67 It is not important. We can leave this as how it is for now. I posted my test result so that you can merge it.

Reply:
ok, thanks, I will set this conversation to resolved.

Reply:
> if (!empty($this->state?->get('params'))) {

Cool. Just tried it while porting a custom extension that was crashing to Joomla 6.